ch_tools/chadmin/internal/table.py

```diff
@@ -249,6 +260,136 @@ def delete_table(
     )
 
 
+def _is_valid_uuid(uuid_str):
+    try:
+        val = uuid.UUID(uuid_str)
+    except ValueError:
+        return False
+    return str(val) == uuid_str
+
+
+def _get_uuid_table(table_metadata_path: str) -> str:
+    table_uuid = ""
+
+    with open(table_metadata_path, "r", encoding="utf-8") as metadata_file:
+        for line in metadata_file:
+            if "UUID" in line:
+                parts = line.split()
+                try:
+                    uuid_index = parts.index("UUID")
+                    # check size
+                    table_uuid = parts[uuid_index + 1].strip("'")
+                except ValueError:
+                    raise RuntimeError(
+                        f"Failed parse UUID from '{table_metadata_path}'"
+                    )
+                break
+        else:
+            raise RuntimeError(f"No UUID in '{table_metadata_path}'")
+
+    assert _is_valid_uuid(table_uuid)
+
+    return table_uuid
+
+
+def check_table_dettached(ctx, database_name, table_name):
+    query = """
+        SELECT
+            1
+        FROM system.tables
+        WHERE database = '{{ database_name }}'
+          AND table = '{{ table_name }}'
+    """
+
+    response = execute_query(
+        ctx,
+        query,
+        database_name=database_name,
+        table_name=table_name,
+        format_="JSON",
+    )["data"]
+
+    if 0 != len(response):
+        raise RuntimeError(
+            f"Table '{database_name}'.'{table_name}' is attach. Use delete without --detach flag."
+        )
+
+
+def _has_object_storage(ctx: Context) -> bool:
+    query = """
+        SELECT 1 FROM system.disks WHERE name = 'object_storage'
```

Comment:
It's better to not depend on concrete disk names. Suggest to select non-local disks by disk type and perform cleanup for them using `clickhouse-disks` tool. Probably we can do the same for local disks too.

Reply:
Ok, Im going to remove logic that relates with disk name.